src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3878,6 +3885,69 @@ public override IOperation VisitTypeOf(ITypeOfOperation operation, int? captureI
         }
 
+        public override IOperation VisitParameterInitializer(IParameterInitializerOperation operation, int? captureIdForResult)
+        {
+            var parameterRef = new ParameterReferenceExpression(operation.Parameter, semanticModel: null,
+                operation.Syntax, operation.Parameter.Type, constantValue: default, isImplicit: true);
+            VisitInitializer(rewrittenTarget: parameterRef, initializer: operation);
+            return null;
+        }
+
+        public override IOperation VisitFieldInitializer(IFieldInitializerOperation operation, int? captureIdForResult)
+        {
+            foreach (IFieldSymbol fieldSymbol in operation.InitializedFields)
+            {
+                IInstanceReferenceOperation instance = fieldSymbol.IsStatic ?
+                    null :
+                    new InstanceReferenceExpression(InstanceReferenceKind.ContainingTypeInstance, semanticModel: null,
+                        operation.Syntax, fieldSymbol.ContainingType, constantValue: default, isImplicit: true);
+                Optional<object> constantValue = fieldSymbol.HasConstantValue ? new Optional<object>(fieldSymbol.ConstantValue) : default;
+                var fieldRef = new FieldReferenceExpression(fieldSymbol, isDeclaration: false, instance, semanticModel: null,
+                    operation.Syntax, fieldSymbol.Type, constantValue, isImplicit: true);
+                VisitInitializer(rewrittenTarget: fieldRef, initializer: operation);
+            }
+
+            return null;
+        }
+
+        public override IOperation VisitPropertyInitializer(IPropertyInitializerOperation operation, int? captureIdForResult)
+        {
+            foreach (IPropertySymbol propertySymbol in operation.InitializedProperties)
+            {
+                var instance = propertySymbol.IsStatic ?
+                    null :
+                    new InstanceReferenceExpression(InstanceReferenceKind.ContainingTypeInstance, semanticModel: null,
+                        operation.Syntax, propertySymbol.ContainingType, constantValue: default, isImplicit: true);
+                var propertyRef = new PropertyReferenceExpression(propertySymbol, instance, arguments: ImmutableArray<IArgumentOperation>.Empty,
+                    semanticModel: null, operation.Syntax, operation.Type, constantValue: default, isImplicit: true);
+                VisitInitializer(rewrittenTarget: propertyRef, initializer: operation);
+            }
+
+            return null;
+        }
+
+        private void VisitInitializer(IOperation rewrittenTarget, ISymbolInitializerOperation initializer)
+        {
+            bool haveLocals = !initializer.Locals.IsEmpty;
+            if (haveLocals)
+            {
+                EnterRegion(new RegionBuilder(ControlFlowGraph.RegionKind.Locals, locals: initializer.Locals));
+            }
+
+            // We skip constants in the control flow graph, as they're not actually involved in any control flow.
```

Comment:
>// We skip constants in the control flow graph, as they're not actually involved in any control flow. [](start = 12, length = 101)

At first this feels strange. Will we proceed with fields that are declared constant, but have non-constant initializer and filter out constant fields that have constant initializer? This feels inconsistent. If filtering constant fields out, then perhaps that should be done on their declaration state, rather than based on the initializer. I would not filter anything out, the fact that initialization of constant fields happens implicitly could be thought of as an implementation detail or an optimization.

Reply:
That seems acceptable - I was trying to be consistent with the approach we tool for const locals, but I like your suggestion.

---
In reply to: [187454869](https://github.com/dotnet/roslyn/pull/26763#discussion_r187454869) [](ancestors = 187454869)